src/main/java/bio/terra/common/stairway/StairwayComponent.java

```diff
@@ -63,24 +68,55 @@ public void initialize(DataSource dataSource, Object context, List<StairwayHook>
     initialize(newStairwayOptionsBuilder().dataSource(dataSource).context(context).hooks(hooks));
   }
 
+  /**
+   * Set up the Stairway work queue. There are two ways we get a work queue. If both the gcp topicId
+   * and subscriptionId are passed as parameters, then we use those. Otherwise, if the
+   * clusterNameSuffix is present, then we create the queue. It is an error to try to run in
+   * Kubernetes without a work queue.
+   */
+  private GcpPubSubQueue setupWorkQueue() {
+    try {
+      String topicId = stairwayProperties.getGcpPubSubTopicId();
+      String subscriptionId = stairwayProperties.getGcpPubSubSubscriptionId();
+      if (topicId == null || subscriptionId == null) {
```

Comment:
optional: if both aren't null, consider asserting that clusterNamePrefix is null to prevent cases where it's not obvious what should happen.

Reply:
I rearranged the logic to test for the two allowed cases and throw with an explanatory error.
I consolidated getClusterName into setupWorkQueue.